homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,236 @@
+"""This component provides basic support for Ezviz IP cameras."""
+import asyncio
+import logging
+
+from haffmpeg.tools import IMAGE_JPEG, ImageFrame
+from pyezviz.camera import EzvizCamera
+from pyezviz.client import EzvizClient, PyEzvizError
+import voluptuous as vol
+
+from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CAMERAS = "cameras"
+
+DEFAULT_CAMERA_USERNAME = "admin"
+DEFAULT_RTSP_PORT = "554"
+
+DATA_FFMPEG = "ffmpeg"
+
+
+EZVIZ_DATA = "ezviz"
+ENTITIES = "entities"
+
+
+CAMERA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): cv.string, vol.Required(CONF_PASSWORD): cv.string}
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_CAMERAS, default={}): {cv.string: CAMERA_SCHEMA},
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, disc_info=None):
+    """Set up the Ezviz IP Cameras."""
+
+    conf_cameras = config[CONF_CAMERAS]
+
+    account = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    try:
+        ezviz_client = EzvizClient(account, password)
+        ezviz_client.login()
+        cameras = ezviz_client.load_cameras()
+
+    except PyEzvizError as exp:
+        _LOGGER.error(exp)
+        return None
+
+    # now, let's build the HASS devices
+    camera_entities = []
+
+    # Add the cameras as devices in HASS
+    for camera in cameras:
+
+        camera_username = DEFAULT_CAMERA_USERNAME
+        camera_password = ""
+        camera_rtsp_stream = ""
+        camera_serial = camera["serial"]
+
+        # There seem to be a bug related to localRtspPort in Ezviz API...
+        local_rtsp_port = DEFAULT_RTSP_PORT
+        if camera["local_rtsp_port"] and camera["local_rtsp_port"] != 0:
+            local_rtsp_port = camera["local_rtsp_port"]
+
+        if camera_serial in conf_cameras:
+            camera_username = conf_cameras[camera_serial]["username"]
```

Comment:
Please use the config key constants we have defined, eg `CONF_USERNAME`, when accessing the config.

Reply:
done